_deploy/r/gnoswap/gnft/gnft_test.gno

```diff
@@ -0,0 +1,450 @@
+		},
+		{
+			name: "transfer token owned by other user",
+			setup: func() {
+				std.TestSetRealm(positionRealm)
+				Mint(a2u(addr01), tid(1))
+			},
+			callerRealm: std.NewUserRealm(addr02),
+			shouldPanic: true,
+			panicMsg:    "caller is not token owner or approved",
+			expected:    std.Address(""),
+		},
+		{
+			name:        "transfer token owned by caller",
+			callerRealm: std.NewUserRealm(addr01),
+			shouldPanic: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.setup != nil {
+				tt.setup()
+			}
+
+			if tt.shouldPanic {
+				uassert.PanicsWithMessage(t, tt.panicMsg, func() {
+					TransferFrom(a2u(addr01), a2u(addr02), tid(1))
+				})
+			} else {
+				std.TestSetRealm(tt.callerRealm)
+				uassert.NotPanics(t, func() {
+					TransferFrom(a2u(addr01), a2u(addr02), tid(1))
+				})
+			}
+		})
+	}
+}
+
+func TestMint(t *testing.T) {
+	resetObject(t)
+
+	tests := []struct {
+		name          string
+		callerRealm   std.Realm
+		tokenIdToMint uint64
+		addressToMint std.Address
+		shouldPanic   bool
+		panicMsg      string
+		expected      string
+	}{
+		{
+			name:        "mint without permission",
+			shouldPanic: true,
+			panicMsg:    "ownable: caller is not owner",
+		},
+		{
+			name:          "mint first nft to addr01",
+			callerRealm:   std.NewCodeRealm(consts.POSITION_PATH),
+			tokenIdToMint: 1,
+			addressToMint: addr01,
+			expected:      "1",
+		},
+		{
+			name:          "mint second nft to addr02",
+			callerRealm:   std.NewCodeRealm(consts.POSITION_PATH),
+			tokenIdToMint: 2,
+			addressToMint: addr02,
+			expected:      "2",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.shouldPanic {
+				uassert.PanicsWithMessage(t, tt.panicMsg, func() {
+					Mint(a2u(tt.addressToMint), tid(tt.tokenIdToMint))
+				})
+
+			} else {
+				std.TestSetRealm(tt.callerRealm)
+				mintedTokenId := Mint(a2u(tt.addressToMint), tid(tt.tokenIdToMint))
+				uassert.Equal(t, tt.expected, string(mintedTokenId))
+			}
+		})
+	}
+}
+
+func TestBurn(t *testing.T) {
+	tests := []struct {
+		name          string
+		callerRealm   std.Realm
+		tokenIdToBurn uint64
+		shouldPanic   bool
+		panicMsg      string
+	}{
+		{
+			name:          "burn without permission",
+			tokenIdToBurn: 1,
+			shouldPanic:   true,
+			panicMsg:      "ownable: caller is not owner",
+		},
+		{
+			name:          "burn non-existent token id",
+			callerRealm:   std.NewCodeRealm(consts.POSITION_PATH),
+			tokenIdToBurn: 99,
+			shouldPanic:   true,
+			panicMsg:      errInvalidTokenId,
+		},
+		{
+			name:          "burn token id(2)",
+			callerRealm:   std.NewCodeRealm(consts.POSITION_PATH),
+			tokenIdToBurn: 2,
+			shouldPanic:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			std.TestSetRealm(tt.callerRealm)
+
+			if tt.shouldPanic {
+				uassert.PanicsWithMessage(t, tt.panicMsg, func() {
+					Burn(tid(tt.tokenIdToBurn))
+				})
```

Comment:
You need to make sure that the tokenList for that address is deleted after the burn.

Reply:
2e25c41db0c6c37365eed4918f2ad05521b8c61f